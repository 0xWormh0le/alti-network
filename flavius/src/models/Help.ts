import UI_STRINGS from 'util/ui-strings'

export interface IHelpQuestion {
  question: string
  risk_ids: RiskTypeId[]
}

export const HelpQuestionsList: IHelpQuestion[] = [
  {
    question: UI_STRINGS.HELP.SHARING_RISKS,
    risk_ids: [1011, 1013, 1020, 1021, 1022, 1023]
  },
  {
    question: UI_STRINGS.HELP.RELATIONSHIP_RISKS,
    risk_ids: [2000, 2001, 2002, 2003]
  },
  {
    question: UI_STRINGS.HELP.ACTIVITY_RISKS,
    risk_ids: [3100, 3200, 3010, 3012]
  },
  {
    question: UI_STRINGS.HELP.INFORMATIONAL_RISKS,
    risk_ids: [0, 10, 11]
  }
]
