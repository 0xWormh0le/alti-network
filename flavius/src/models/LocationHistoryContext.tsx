import React from 'react'
import pick from 'lodash/pick'
import { noop } from 'lodash'

const TRACKED_PARAMS = [
  'page',
  'pageNumber',
  'riskTypeIds',
  'selectedCardKey',
  'selectedSubNavKey',
  'selectedEmail',
  'platformIds',
  'userVisibilityState'
]

export const LocationHistoryContext = React.createContext({} as LocationHistoryContextProviderState)

interface LocationHistoryContextProviderProps {
  children: React.ReactNode
}

interface LocationHistoryContextProviderState {
  breadcrumbLocations: string[]
  trackedQueryParams: {}
  deleteFromBreadcrumbLocations?: (path: string) => void
  onLocationVisited?: (location: string, breadcrumbPosition?: number) => void
  onQueryParamsChanged?: (queryParams: object) => void
}

export class LocationHistoryContextProvider extends React.Component<
  LocationHistoryContextProviderProps,
  LocationHistoryContextProviderState
> {
  public state = {
    breadcrumbLocations: [], // array of the last location visited
    trackedQueryParams: {}, // keep track of the latest value for some interesting query params before opening modals
    deleteFromBreadcrumbLocations: noop
  }

  public render() {
    return (
      <LocationHistoryContext.Provider
        value={{
          breadcrumbLocations: this.state.breadcrumbLocations,
          deleteFromBreadcrumbLocations: (path) => {
            this.setState((prevState) => {
              return {
                breadcrumbLocations: prevState.breadcrumbLocations.slice(0, prevState.breadcrumbLocations.indexOf(path))
              }
            })
          },
          onLocationVisited: (location: string, breadcrumbPosition?: number) => {
            if (breadcrumbPosition === undefined) {
              // push new location to the back of the array and just keep the last 3
              this.setState((prevState) => ({
                breadcrumbLocations: [...prevState.breadcrumbLocations, location].slice(-3)
              }))
            } else {
              // this means the user clicked on the breadcrumb and navigated to a breadcrumb path,
              // we need to reset the locations stack in that case
              this.setState((prevState) => ({
                breadcrumbLocations: prevState.breadcrumbLocations.slice(0, breadcrumbPosition + 1)
              }))
            }
          },
          trackedQueryParams: this.state.trackedQueryParams,
          onQueryParamsChanged: (queryParams: object) => {
            this.setState({
              trackedQueryParams: pick(queryParams, TRACKED_PARAMS)
            })
          }
        }}>
        {this.props.children}
      </LocationHistoryContext.Provider>
    )
  }
}

// Consumer
export const LocationHistoryContextConsumer = LocationHistoryContext.Consumer
