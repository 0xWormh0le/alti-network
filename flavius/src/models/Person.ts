import { UserKind, AccessLevel } from 'types/common'
import UI_STRINGS from 'util/ui-strings'
import anonymousUserIcon from 'icons/anonymous-user.svg'
class Person implements IPerson {
  public accessLevel: AccessLevel
  public name: Name
  public projectId: string
  public primaryEmail: Email
  public userKind: UserKind
  public altnetId?: string
  public recoveryEmail?: Email
  public phones: Phone[]
  public externalIds: UserId[]
  public avatar?: Avatar
  public internal?: boolean // deprecated thanks to userKind
  public anonymous?: boolean
  public internalCount?: number
  public externalCount?: number
  public emails: Email[]
  public accessCount?: number
  public riskCount?: number
  public orgUnitPath?: string
  public isEnrolledInMFA?: boolean
  public etag?: string
  public creationTime?: number
  public lastLoginTime?: number
  public lastModifiedTime?: number
  public notes?: Notes
  public lastRemovedPermissions?: number | null
  public title?: string
  public department?: string
  public phoneNumber?: string

  constructor(person: IPerson) {
    this.altnetId = person.altnetId
    this.projectId = person.projectId
    this.userKind = person.userKind
    this.accessLevel = person.accessLevel
    this.name = person.name || {
      givenName: '',
      familyName: '',
      fullName: ''
    }
    this.primaryEmail = person.primaryEmail
    this.recoveryEmail = person.recoveryEmail
    this.phones = person.phones || []
    this.externalIds = person.externalIds || []
    this.orgUnitPath = person.orgUnitPath
    this.etag = person.etag
    this.isEnrolledInMFA = person.isEnrolledInMFA
    this.creationTime = person.creationTime
    this.lastLoginTime = person.lastLoginTime
    this.lastModifiedTime = person.lastModifiedTime
    this.notes = person.notes
    this.avatar =
      this.name.fullName === UI_STRINGS.RISKS.ANONYMOUS_USER ? { url: anonymousUserIcon, url_etag: '' } : person.avatar
    this.title = person.title
    this.department = person.department
    this.phoneNumber = person.phoneNumber
    this.internal = person.internal
    this.anonymous = !person.primaryEmail?.address || person.primaryEmail?.address?.indexOf('@') === -1 // altnetId: person.personId?.indexOf('@') === -1 &&
    this.internalCount = person.internalCount
    this.externalCount = person.externalCount
    this.emails = person.emails || []
    this.accessCount = person.accessCount
    this.riskCount = person.riskCount
    this.lastRemovedPermissions = person.lastRemovedPermissions
  }

  public get displayName(): string {
    if (this.name.givenName && this.name.familyName) {
      return `${this.name.givenName} ${this.name.familyName}`
    } else if (this.anonymous) {
      return UI_STRINGS.RISKS.ANONYMOUS_USER
    } else if (this.primaryEmail?.address) {
      const emailFirst = this.primaryEmail?.address.split('@')
      return emailFirst && emailFirst.length > 0 ? emailFirst[0] : this.primaryEmail?.address
    } else {
      return this.primaryEmail?.address ?? '' // altnetId:
    }
  }

  /*
   * For transform value from Map to Object in griddle-react custom components, e.g. SimplePersonCell, ActorCell
   * This is a workaround solution, once we refactor the table to NOT use griddle-react, this method shall be gone
   */
  static fromMap(personMap: { get: (x: keyof IPerson) => any }): Person {
    return new Person({
      primaryEmail: personMap.get('primaryEmail') && (Object.fromEntries(personMap.get('primaryEmail')) as Email),
      recoveryEmail: personMap.get('recoveryEmail') && (Object.fromEntries(personMap.get('recoveryEmail')) as Email),
      avatar: personMap.get('avatar') && (Object.fromEntries(personMap.get('avatar')) as Avatar),
      name: personMap.get('name') && (Object.fromEntries(personMap.get('name')) as Name),
      internal: personMap.get('internal'),
      internalCount: personMap.get('internalCount'),
      externalCount: personMap.get('externalCount'),
      emails:
        personMap.get('emails') &&
        (Array.from(personMap.get('emails')).map((email: any) => Object.fromEntries(email)) as Email[]),
      accessCount: personMap.get('accessCount'),
      riskCount: personMap.get('riskCount'),
      altnetId: personMap.get('altnetId'),
      projectId: personMap.get('projectId'),
      accessLevel: personMap.get('accessLevel'),
      userKind: personMap.get('userKind'),
      phones:
        personMap.get('phones') &&
        (Array.from(personMap.get('phones')).map((email: any) => Object.fromEntries(email)) as Phone[]),
      externalIds:
        personMap.get('externalIds') &&
        (Array.from(personMap.get('externalIds')).map((email: any) => Object.fromEntries(email)) as UserId[]),
      orgUnitPath: personMap.get('orgUnitPath'),
      etag: personMap.get('etag'),
      isEnrolledInMFA: personMap.get('isEnrolledInMFA'),
      creationTime: personMap.get('creationTime'),
      lastLoginTime: personMap.get('lastLoginTime'),
      lastModifiedTime: personMap.get('lastModifiedTime'),
      notes: personMap.get('notes') && (Object.fromEntries(personMap.get('notes')) as Notes)
    })
  }
}

export default Person
