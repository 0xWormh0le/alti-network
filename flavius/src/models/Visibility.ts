import compact from 'lodash/compact'
import { isDemoEnv, isDevEnv } from 'util/helpers'
import {
  activityItem,
  dashboardItem,
  risksItem,
  settingsItem,
  searchItem,
  riskSettingsItem,
  managePlatformsItem
} from 'components/base/NavigationItems'

type NullableAuthenticatedUser = AuthenticatedUser | null

// This is used to map visibility levels to the custom user attributes defined in Amazon
// Cognito, look for `visibility_level` on the User Pools.

export const VISIBILITY_LEVEL_ATTRIBUTE = 'custom:visibility_level'

export const VisibilityLevel = {
  HIGH: 'high',
  MEDIUM: 'medium',
  LOW: 'low'
}

const demoEnvironment = isDemoEnv
// temp hide box lite features/routes for staging and production build ONLY, will remove later once box lite is ready
const devEnvironment = isDevEnv

const items = demoEnvironment
  ? [dashboardItem, risksItem, searchItem]
  : devEnvironment
  ? [dashboardItem, risksItem, searchItem, riskSettingsItem, activityItem]
  : [dashboardItem, risksItem, searchItem, riskSettingsItem]

const navigationItemsVisibilityLevelMap = {
  [VisibilityLevel.HIGH]: items,
  [VisibilityLevel.MEDIUM]: items,
  [VisibilityLevel.LOW]: items
}

const supportItemsVisibilityLevelMap = devEnvironment
  ? {
      // settings page is disabled on Demo Environment
      [VisibilityLevel.HIGH]: compact([!demoEnvironment && settingsItem, managePlatformsItem]),
      [VisibilityLevel.MEDIUM]: compact([!demoEnvironment && settingsItem, managePlatformsItem]),
      [VisibilityLevel.LOW]: compact([!demoEnvironment && settingsItem, managePlatformsItem])
    }
  : {
      // settings page is disabled on Demo Environment
      [VisibilityLevel.HIGH]: compact([!demoEnvironment && settingsItem]),
      [VisibilityLevel.MEDIUM]: compact([!demoEnvironment && settingsItem]),
      [VisibilityLevel.LOW]: compact([!demoEnvironment && settingsItem])
    }

function getVisibilityFromAuthenticatedUser(user: NullableAuthenticatedUser) {
  // default visibility: we provide access to all navigation elements (HIGH)
  let visibility = VisibilityLevel.HIGH

  if (user && user.attributes && user.attributes[VISIBILITY_LEVEL_ATTRIBUTE]) {
    visibility = user.attributes[VISIBILITY_LEVEL_ATTRIBUTE]
  }

  return visibility
}

// Based on the visibility level of the authenticatedUser, this will return the menu
// items and routes that the user is allowed to access
export function getNavigationItems(authenticatedUser: NullableAuthenticatedUser) {
  if (!authenticatedUser) {
    return []
  }

  const visibility = getVisibilityFromAuthenticatedUser(authenticatedUser)
  return navigationItemsVisibilityLevelMap[visibility]
}

// Based on the visibility level of the authenticatedUser, this will return the support menu
// items and routes that the user is allowed to access
export function getSupportItems(authenticatedUser: NullableAuthenticatedUser) {
  if (!authenticatedUser) {
    return []
  }

  const visibility = getVisibilityFromAuthenticatedUser(authenticatedUser)
  return supportItemsVisibilityLevelMap[visibility]
}
