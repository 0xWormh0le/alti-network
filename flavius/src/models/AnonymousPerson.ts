import Person from './Person'
import { UsageKind } from 'types/common'

const email: Email = {
  address: 'sample@foo.bar',
  kind: UsageKind.personal,
  primary: false,
  accessCount: 1,
  riskCount: 1,
  lastDeletedPermissions: 0
}

class AnonymousPerson extends Person {
  constructor() {
    super({
      name: {
        familyName: 'User',
        givenName: 'Anonymous',
        fullName: 'User Anonymous'
      },
      primaryEmail: email
    } as IPerson)
  }
}

export default AnonymousPerson
