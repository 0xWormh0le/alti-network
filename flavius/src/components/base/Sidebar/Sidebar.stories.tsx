import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import FlaviusSidebar from './Sidebar'
import { BrowserRouter } from 'react-router-dom'
import { UserContext } from 'models/UserContext'

export default {
  title: 'Base/Sidebar',
  component: FlaviusSidebar
} as Meta

const Template: Story<any> = (args) => (
  <BrowserRouter>
    <UserContext.Provider
      value={{
        domains: [],
        isTrial: false,
        userHasAuthenticated: (user) => null,
        setShowWelcomeDialog: (set) => null,
        showWelcomeDialog: true,
        authenticatedUser: {
          username: 'Bobbie',
          id: '111',
          attributes: {
            email: 'bobbie@thoughtlabs.io',
            email_verified: true,
            name: 'Bobbie Hawaii',
            phoneNumber: '4567890',
            phoneNumber_verified: true,
            sub: 'Sub'
          }
        }
      }}>
      <FlaviusSidebar {...args} />;
    </UserContext.Provider>
  </BrowserRouter>
)

export const Default = Template.bind({})
Default.args = {}
