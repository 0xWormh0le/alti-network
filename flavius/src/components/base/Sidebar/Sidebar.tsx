import React from 'react'
import { Link, useLocation } from 'react-router-dom'
import { generateRoutePath } from 'util/helpers'
import Tooltip from 'components/widgets/Tooltip'
// import BaseIcon from 'icons/BaseIcon'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import { getNavigationItems, getSupportItems } from 'models/Visibility'
import { UserContextConsumer } from 'models/UserContext'
import UI_STRINGS from 'util/ui-strings'
import { Sidebar, SidebarSection, SidebarItem, SidebarNavMenu } from '@altitudenetworks/component-library'
import '@altitudenetworks/component-library/dist/scss/components/elements/Sidebar.scss'

const RouteItem = (props: { item: any; pathname: string }) => {
  const { item, pathname } = props

  if (item.disabled)
    return (
      <Tooltip text={UI_STRINGS.SIDEBAR.PAGE_NOT_ACTIVATED} id={`tooltip-sidebar-disabled-${item.label}-id`}>
        <div className='Sidebar__item Sidebar__link Sidebar__link--disabled'>
          {item.icon}
          <Typography variant={TypographyVariant.LABEL} weight='medium' className='align-middle'>
            {item.label}
          </Typography>
        </div>
      </Tooltip>
    )

  return (
    <Link className='Sidebar__link' to={generateRoutePath(item.routePath || item.label)}>
      <SidebarItem key={item.label} Icon={item.icon} isActive={pathname === generateRoutePath(item.label)}>
        {item.label}
      </SidebarItem>
    </Link>
  )
}

const MainSidebar: React.FC = () => {
  const { pathname } = useLocation()

  return (
    <UserContextConsumer>
      {({ authenticatedUser }) => (
        <Sidebar>
          <SidebarSection title={UI_STRINGS.SIDEBAR.MAIN_MENU}>
            {getNavigationItems(authenticatedUser).map((item) =>
              item.subItems && item.subItems.length ? (
                <SidebarNavMenu key={item.key} title={item.label} Icon={item.icon}>
                  {item.subItems.map((subItem) => (
                    <RouteItem key={subItem.key} item={subItem} pathname={pathname} />
                  ))}
                </SidebarNavMenu>
              ) : (
                <RouteItem key={item.key} item={item} pathname={pathname} />
              )
            )}
          </SidebarSection>

          <SidebarSection title={UI_STRINGS.SIDEBAR.SUPPORT}>
            {getSupportItems(authenticatedUser).map((item) => (
              <RouteItem key={item.label} item={item} pathname={pathname} />
            ))}
          </SidebarSection>
        </Sidebar>
      )}
    </UserContextConsumer>
  )
}

export default MainSidebar
