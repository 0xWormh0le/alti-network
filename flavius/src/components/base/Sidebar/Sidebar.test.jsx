import React from 'react'
import { renderWithRouter } from 'test/support/helpers'
import { UserContextProvider } from 'models/UserContext'
import Sidebar from './Sidebar'
import { VisibilityLevel, VISIBILITY_LEVEL_ATTRIBUTE } from 'models/Visibility'

const userNoVisibilityLevel = {
  attributes: {
    name: 'John Doe'
  }
}

const userVisibilityLevelHigh = {
  attributes: {
    name: 'John Doe',
    [VISIBILITY_LEVEL_ATTRIBUTE]: VisibilityLevel.HIGH
  }
}

const userVisibilityLevelMedium = {
  attributes: {
    name: 'John Doe',
    [VISIBILITY_LEVEL_ATTRIBUTE]: VisibilityLevel.MEDIUM
  }
}

const userVisibilityLevelLow = {
  attributes: {
    name: 'John Doe',
    [VISIBILITY_LEVEL_ATTRIBUTE]: VisibilityLevel.LOW
  }
}

function renderSidebar(authenticatedUser) {
  return (
    <UserContextProvider value={{ authenticatedUser }}>
      <Sidebar />
    </UserContextProvider>
  )
}

// Basic example use on how to do snapshot testing on components
it('renders correctly with a user with no visibility level', () => {
  const { container } = renderWithRouter(renderSidebar(userNoVisibilityLevel))
  expect(container.firstChild).toMatchSnapshot()
})

describe('Visibility levels', () => {
  it('renders all menu items when the visibility level is `high`', () => {
    const { container } = renderWithRouter(renderSidebar(userVisibilityLevelHigh))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders some menu items when the visibility level is `medium`', () => {
    const { container } = renderWithRouter(renderSidebar(userVisibilityLevelMedium))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders some menu items when the visibility level is `low`', () => {
    const { container } = renderWithRouter(renderSidebar(userVisibilityLevelLow))
    expect(container.firstChild).toMatchSnapshot()
  })
})
