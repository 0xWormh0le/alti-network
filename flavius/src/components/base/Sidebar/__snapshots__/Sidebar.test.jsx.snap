// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visibility levels renders all menu items when the visibility level is \`high\` 1`] = `
<nav
  class="Sidebar "
>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Main menu
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/dashboard"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              dashboard.svg
            </svg>
          </span>
          <span>
            Dashboard
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risks"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              iron.svg
            </svg>
          </span>
          <span>
            Risks
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/search"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              search.svg
            </svg>
          </span>
          <span>
            Search
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risk-settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              risk-settings.svg
            </svg>
          </span>
          <span>
            Risk Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Support
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              settings.svg
            </svg>
          </span>
          <span>
            Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
</nav>
`;

exports[`Visibility levels renders some menu items when the visibility level is \`low\` 1`] = `
<nav
  class="Sidebar "
>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Main menu
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/dashboard"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              dashboard.svg
            </svg>
          </span>
          <span>
            Dashboard
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risks"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              iron.svg
            </svg>
          </span>
          <span>
            Risks
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/search"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              search.svg
            </svg>
          </span>
          <span>
            Search
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risk-settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              risk-settings.svg
            </svg>
          </span>
          <span>
            Risk Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Support
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              settings.svg
            </svg>
          </span>
          <span>
            Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
</nav>
`;

exports[`Visibility levels renders some menu items when the visibility level is \`medium\` 1`] = `
<nav
  class="Sidebar "
>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Main menu
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/dashboard"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              dashboard.svg
            </svg>
          </span>
          <span>
            Dashboard
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risks"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              iron.svg
            </svg>
          </span>
          <span>
            Risks
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/search"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              search.svg
            </svg>
          </span>
          <span>
            Search
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risk-settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              risk-settings.svg
            </svg>
          </span>
          <span>
            Risk Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Support
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              settings.svg
            </svg>
          </span>
          <span>
            Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
</nav>
`;

exports[`renders correctly with a user with no visibility level 1`] = `
<nav
  class="Sidebar "
>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Main menu
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/dashboard"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              dashboard.svg
            </svg>
          </span>
          <span>
            Dashboard
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risks"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              iron.svg
            </svg>
          </span>
          <span>
            Risks
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/search"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              search.svg
            </svg>
          </span>
          <span>
            Search
          </span>
        </div>
      </a>
      <a
        class="Sidebar__link"
        href="/risk-settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              risk-settings.svg
            </svg>
          </span>
          <span>
            Risk Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
  <div
    class="Sidebar__section"
  >
    <div
      class="Typography Typography--subhead1 Typography--uppercase Sidebar__group-title"
    >
      Support
    </div>
    <ul
      class="Sidebar__list"
    >
      <a
        class="Sidebar__link"
        href="/settings"
      >
        <div
          class="Sidebar__item "
        >
          <span
            class="Sidebar__icon"
          >
            <svg>
              settings.svg
            </svg>
          </span>
          <span>
            Settings
          </span>
        </div>
      </a>
    </ul>
  </div>
</nav>
`;
